### ISSUE TITLE
Requests sometimes hang when a backend isn't responding

### ISSUE THREAD

**Issue by BlackYoup** [sozu-proxy/sozu#475]
I'm having a weird problem where my requests will hang if one of my application's backend is down.

On the first request, it will correctly serve the response. On the second request the `error connecting to backend, trying again` message will appear and it will stay here forever.

To reproduce, configure it to have an application with two backends:
```
[applications.MyApp]
hostname = "lolcatho.st"

[[applications.MyApp.backends]]
address = "127.0.0.1:1026"
# used as load balancing parameter
weight = 100
# value of the sticky session cookie
sticky_id = "backend-1026"

[[applications.MyApp.backends]]
address = "127.0.0.1:1027"
# used as load balancing parameter
weight = 100
# value of the sticky session cookie
sticky_id = "backend-1027"
```

Then, do multiple curls: `while true; do curl lolcatho.st:8080; done`. The second curl should hang and trigger the problem.

If there is only one backend, it will correctly serve the response (200 if UP, 503 if down).

Tcpdump shows that it correctly connects to the backend but no network activity after that:
```
# first request
00:24:21.554719 IP localhost.58276 > localhost.1026: Flags [S], seq 4134314193, win 43690, options [mss 65495,sackOK,TS val 1856008874 ecr 0,nop,wscale 7], length 0
00:24:21.554744 IP localhost.1026 > localhost.58276: Flags [S.], seq 1575486492, ack 4134314194, win 43690, options [mss 65495,sackOK,TS val 1856008874 ecr 1856008874,nop,wscale 7], length 0
00:24:21.554765 IP localhost.58276 > localhost.1026: Flags [.], ack 1, win 342, options [nop,nop,TS val 1856008874 ecr 1856008874], length 0
00:24:21.555144 IP localhost.58276 > localhost.1026: Flags [P.], seq 1:79, ack 1, win 342, options [nop,nop,TS val 1856008874 ecr 1856008874], length 78
00:24:21.555167 IP localhost.1026 > localhost.58276: Flags [.], ack 79, win 342, options [nop,nop,TS val 1856008874 ecr 1856008874], length 0
00:24:21.555206 IP localhost.58276 > localhost.1026: Flags [P.], seq 79:260, ack 1, win 342, options [nop,nop,TS val 1856008874 ecr 1856008874], length 181
00:24:21.555223 IP localhost.1026 > localhost.58276: Flags [.], ack 260, win 350, options [nop,nop,TS val 1856008874 ecr 1856008874], length 0
00:24:21.555254 IP localhost.58276 > localhost.1026: Flags [P.], seq 260:262, ack 1, win 342, options [nop,nop,TS val 1856008874 ecr 1856008874], length 2
00:24:21.555267 IP localhost.1026 > localhost.58276: Flags [.], ack 262, win 350, options [nop,nop,TS val 1856008874 ecr 1856008874], length 0
00:24:21.555839 IP localhost.1026 > localhost.58276: Flags [P.], seq 1:123, ack 262, win 350, options [nop,nop,TS val 1856008875 ecr 1856008874], length 122
00:24:21.555859 IP localhost.58276 > localhost.1026: Flags [.], ack 123, win 342, options [nop,nop,TS val 1856008875 ecr 1856008875], length 0
00:24:21.558364 IP localhost.58276 > localhost.1026: Flags [F.], seq 262, ack 123, win 342, options [nop,nop,TS val 1856008877 ecr 1856008875], length 0
00:24:21.558701 IP localhost.1026 > localhost.58276: Flags [F.], seq 123, ack 263, win 350, options [nop,nop,TS val 1856008877 ecr 1856008877], length 0
00:24:21.558731 IP localhost.58276 > localhost.1026: Flags [.], ack 124, win 342, options [nop,nop,TS val 1856008878 ecr 1856008877], length 0
# end of first request
# second request
00:24:22.956571 IP localhost.58282 > localhost.1026: Flags [S], seq 4021667740, win 43690, options [mss 65495,sackOK,TS val 1856010275 ecr 0,nop,wscale 7], length 0
00:24:22.956604 IP localhost.1026 > localhost.58282: Flags [S.], seq 2081279961, ack 4021667741, win 43690, options [mss 65495,sackOK,TS val 1856010275 ecr 1856010275,nop,wscale 7], length 0
00:24:22.956635 IP localhost.58282 > localhost.1026: Flags [.], ack 1, win 342, options [nop,nop,TS val 1856010275 ecr 1856010275], length 0
# it hangs after SYN ACK
```

Sozu logs: https://gist.github.com/BlackYoup/2940da5b851df990ae114f2323d50a70

**Comment by Geal** [comment]
this should be fixed by b502d5f
(explanation in the commit message)

**Comment by BlackYoup** [comment]
It works now :+1:
